### PR TITLE
move to spark 3.1.1 fixed schema inference issue on write

### PR DESCRIPTION
**Pull request by moderakh**
move to spark 3.1.1 
Due to a bug in spark 3.0.x schema-inference used to be invoked even for write operation. after moving to spark 3.1.1 this is resolved (I verified this).

I still kept `ACCEPT_ANY_SCHEMA` as a capability @ealsur  do you see any reason `ACCEPT_ANY_SCHEMA` need to be removed?

TODO: we should change the folder structure to reflect the artifactId change `sdk/cosmos/azure-cosmos-spark_3-0_2-12` -> `sdk/cosmos/azure-cosmos-spark_3-1_2-12`. I am not going to do that for now because it will create unnecessary git merge issues for others working on the project (we will do later).

### CODE REVIEW

**Review by FabianMeiswinkel — APPROVED** [review]
LGTM

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala [hunk 1]
```diff
@@ -216,33 +216,37 @@ private case class CosmosSchemaInferenceConfig(inferSchemaSamplingSize: Int,
...
+
+  private val inferSchemaSamplingSize = CosmosConfigEntry[Int](key = "spark.cosmos.read.inferSchemaSamplingSize",
+    mandatory = false,
+    defaultValue = Some(DefaultSampleSize),
+    parseFromStringFunction = size => size.toInt,
+    helpMessage = "Sampling size to use when inferring schema")
+
+  private val inferSchemaEnabled = CosmosConfigEntry[Boolean](key = "spark.cosmos.read.inferSchemaEnabled",
+    mandatory = false,
+    defaultValue = Some(true),
```
Comment (ealsur) [inline sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/CosmosConfig.scala:229]: Nice catch, when I wrote this I didn't understand the defaults :)

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ItemsTable.scala [hunk 2]
```diff
@@ -67,8 +67,6 @@ private class ItemsTable(val sparkSession: SparkSession,
   override def name(): String = tableName
 
   override def capabilities(): util.Set[TableCapability] = Set(
-    // ACCEPT_ANY_SCHEMA is needed because of this bug https://github.com/apache/spark/pull/30273
-    // It was fixed in Spark 3.1.0 but Databricks currently only supports 3.0.1
     TableCapability.ACCEPT_ANY_SCHEMA,
```
Comment (ealsur) [inline sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ItemsTable.scala:70]: Is this needed now? If we are using Spark 3.1.1?

**Review by ealsur — APPROVED** [review]

File: sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ItemsTable.scala (hunk 2 above)
Comment (moderakh) [reply sdk/cosmos/azure-cosmos-spark_3-0_2-12/src/main/scala/com/azure/cosmos/spark/ItemsTable.scala:70]: as per javadoc:

`ACCEPT_ANY_SCHEMA` Signals that the table accepts input of any schema in a write operation.

https://spark.apache.org/docs/3.1.1/api/java/org/apache/spark/sql/connector/catalog/TableCapability.html#ACCEPT_ANY_SCHEMA

As per its description it seems its purpose is something else and not relevant to the bug in spark 3.0, that's why I kept it.

**Pull request merged**